lib/SILGen/SILGenExpr.cpp

```diff
@@ -2357,9 +2357,37 @@ RValue RValueEmitter::visitCaptureListExpr(CaptureListExpr *E, SGFContext C) {
   return visit(E->getClosureBody(), C);
 }
 
+static OpaqueValueExpr *opaqueValueExprToSubstituteForAutoClosure(
+    const AbstractClosureExpr *e) {
+  // When we find an autoclosure that just calls an opaque closure,
+  // this is a case where we've created the opaque closure as a
+  // stand-in for the autoclosure itself. Such an opaque closure is
+  // created when we have a property wrapper's 'init(wrappedValue:)'
+  // taking an autoclosure argument.
+  if (auto ace = dyn_cast<AutoClosureExpr>(e)) {
+    if (auto ce = dyn_cast<CallExpr>(ace->getSingleExpressionBody())) {
+      if (auto ove = dyn_cast<OpaqueValueExpr>(ce->getFn())) {
+        if (!ace->isImplicit() || !ove->isImplicit() || !ove->isPlaceholder())
+          return nullptr;
+
+        Type aceType = ace->getType();
+        Type oveType = ove->getType();
+        if (!aceType || !oveType)
+          return nullptr;
+
+        if (aceType->getCanonicalType() == oveType->getCanonicalType())
+          return ove;
+      }
+    }
+  }
+  return nullptr;
+}
 
 RValue RValueEmitter::visitAbstractClosureExpr(AbstractClosureExpr *e,
                                                SGFContext C) {
+  if (auto ove = opaqueValueExprToSubstituteForAutoClosure(e))
+    return visitOpaqueValueExpr(ove, C);
```

Comment:
It's not clear if this is an optimization or has semantic effect. Can you add a clarifying comment?